lib/protocol/http/body/readable.rb

```diff
@@ -83,11 +83,15 @@ def finish
 				
 				# Enumerate all chunks until finished, then invoke `#close`.
 				def each
-					while chunk = self.read
-						yield chunk
-					end
-				ensure
-					self.close($!)
+          return to_enum(:each) unless block_given?
+					
+          begin
+            while chunk = self.read
+              yield chunk
+            end
+          ensure
+            self.close($!)
+          end
```

Comment:
A couple style comments (feel free to change):

- `to_enum` does not need `:each` argument, as that's the default first argument. I left it because it looks more readable this way.
- In order to make `to_enum(:each)` work I needed to wrap method body in `begin..ensure`. This way `self.close` is called after enumerator is done. Otherwise the `ensure` block would be called immediately after `to_enum(:each)` is called.